signer/src/stacks/api.rs

```diff
@@ -663,6 +670,44 @@ impl StacksClient {
             .map_err(Error::UnexpectedStacksResponse)
     }
 
+    /// Get information about the sortition related to a consensus hash.
+    ///
+    /// Uses the GET /v3/sortitions stacks node endpoint for retrieving
+    /// sortition information.
+    #[tracing::instrument(skip(self))]
+    pub async fn get_sortition_info(
+        &self,
+        consensus_hash: &ConsensusHash,
+    ) -> Result<SortitionInfo, Error> {
+        let path = format!("/v3/sortitions/consensus/{}", consensus_hash);
+        let url = self
+            .endpoint
+            .join(&path)
+            .map_err(|err| Error::PathJoin(err, self.endpoint.clone(), Cow::Owned(path)))?;
+
+        tracing::debug!("Making request to the stacks node for sortition info");
+        let response = self
+            .client
+            .get(url.clone())
+            .timeout(REQUEST_TIMEOUT)
+            .send()
+            .await
+            .map_err(Error::StacksNodeRequest)?;
+
+        response
+            .error_for_status()
+            .map_err(Error::StacksNodeResponse)?
+            .json::<Vec<SortitionInfo>>()
+            .await
+            .map_err(Error::UnexpectedStacksResponse)
+            .and_then(|result| {
+                result
+                    .into_iter()
+                    .next()
+                    .ok_or(Error::InvalidStacksResponse("missing sortition info"))
+            })
```

Comment:
Ahhh, so the stacks-core response is a list with one element in our case. If we asked for the "latest-and-last" then we will receive a list with two elements. Well, at least that's the behavior [on main](https://github.com/stacks-network/stacks-core/blob/40059a57cd27e740c5e9d91a833fb2c975b0bf0b/stackslib/src/net/api/getsortition.rs#L333-L369). I'd add a comment with a link to the source somewhere around here stating that the vector should have exactly one element since we supply the consensus hash.